src/settings/SettingsModal.tsx

```diff
@@ -0,0 +1,254 @@
+import { register, unregister } from '@tauri-apps/api/globalShortcut';
+import { useEffect, useState } from 'react';
+import { VscClose } from 'react-icons/vsc';
+
+import { JSONDebug, JSONDebugContainer } from '../components/JSONDebug';
+import { cx } from '../cx';
+import { useCallablePromise } from '../hooks/useCallablePromise';
+import { getSettings, initSettings, setSettings } from './settings';
+
+// Tauri HOTKEYS
+// https://github.com/tauri-apps/global-hotkey/blob/0b91f4beb998526103447d890ed8eeddc0397b7d/src/hotkey.rs#L164
+const SETTINGS_SHORTCUT_TOGGLE = 'Cmd+,';
+const SETTINGS_SHORTCUT_CLOSE = 'Esc'; // 'Shift+Esc';
+
+type SettingsPaneId = 'user-account' | 'project-general' | 'project-openai' | 'debug';
+interface PaneConfig {
+  id: SettingsPaneId;
+  title: string;
+  Pane: React.FC<{ config: PaneConfig }>;
+}
+
+interface SettingsPanesConfig {
+  section: string;
+  bottom?: boolean;
+  hidden?: boolean;
+  panes: PaneConfig[];
+}
+
+const SETTINGS_PANES: SettingsPanesConfig[] = [
+  {
+    section: 'User',
+    panes: [
+      {
+        id: 'user-account',
+        title: 'User Account',
+        Pane: ToDoSettingsPane,
+      },
+    ],
+  },
+  {
+    section: 'Project',
+    panes: [
+      {
+        id: 'project-general',
+        title: 'General',
+        Pane: ToDoSettingsPane,
+      },
+      {
+        id: 'project-openai',
+        title: 'Open AI',
+        Pane: OpenAiSettingsPane,
+      },
+    ],
+  },
+  {
+    section: 'Debug',
+    bottom: true,
+    hidden: !import.meta.env.DEV,
+    panes: [
+      {
+        id: 'debug',
+        title: 'Config',
+        Pane: DebugSettingsPane,
+      },
+    ],
+  },
+];
+
+// Ensure settings are configured and loaded
+await initSettings();
+
+export function SettingsModal({ open = false, onToggle }: { open?: boolean; onToggle(open: boolean): void }) {
+  const [pane, selectPane] = useState<PaneConfig>(
+    SETTINGS_PANES.flatMap((e) => e.panes).find((p) => p.id === 'project-openai')!,
+  );
+
+  useEffect(() => {
+    (async function runAsync() {
+      try {
+        await unregister(SETTINGS_SHORTCUT_TOGGLE);
+        await unregister(SETTINGS_SHORTCUT_CLOSE);
+        await register(SETTINGS_SHORTCUT_TOGGLE, () => onToggle(!open));
+        await register(SETTINGS_SHORTCUT_CLOSE, () => onToggle(false));
```

Comment:
Presumably you don't want <kbd>ESC</kbd> to be registered globally, only when the Settings modal is open?

My suggestion would be:

- `SettingsModal` is always open when it's in the virtual DOM.
- Its visibility is owned by the parent (`App`) which also registers the <kbd>Cmd</kbd><kbd>,</kbd> shortcut (and presumably never needs to unregister it).
- `SettingsModal` registers the <kbd>ESC</kbd> shortcut and unregisters it when it unmounts. It can just have an `onClose` prop.

Reply:
My (initial) idea was to have a drop-in component for the settings that handles the open/close behaviour. Meanwhile, with the sidebar action to open the settings I've have to move the open/close control to the App.

With the addition of the window menu entries and handles I rewrite this component to:
 - Stop using Tauri's global shortcut (that feature is for scenarios when you want a OS-wide shortcut trigger for the app)
 - Adopt Tauri's [events](https://tauri.app/v1/guides/features/events) for the communication between Rust and the front-end
 - New event for the settings menu click (tauri://menu/settings)
 - Emit settings menu event when on clicks in the Settings icon in the sidebar